Sources/StreamChat/Utils/NotificationExtension+Lifecycle.swift

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+class NotificationExtensionLifecycle {
```

Comment:
While I completely agree, it would not be consistent with the rest of our codebase.
This is something that we might be needing to discuss in the tech discussion and plan a full move to protocols + final classes.